### PR TITLE
Backoffice order, using stripe payment: when card requires auth, send authorization email to guest user

### LINKED ISSUE #8475: Backoffice orders: guest customers do not receive Stripe SCA authorization emails
## Description

For backoffice orders: the email requesting payment authorization for a Stripe SCA payment is not arriving, if the customer is not registered (guest customer).

This concerns nr. 19 in [the list ](https://github.com/openfoodfoundation/openfoodnetwork/wiki/Emails) 

This bug, together with https://github.com/openfoodfoundation/openfoodnetwork/issues/7905 prevents the customer from providing authorization for Stripe SCA payments (for cards requiring it).

## Expected Behavior

Guest customers should receive an Email with link to provide payment authorization, when orders are placed from the BO.

## Actual Behaviour

Guest customers are not receiving the Email with link to provide payment authorization, when orders are placed from the BO.

## Steps to Reproduce

1. Login as a hub admin, which has Stripe SCA payment method set
2. Place an order for a guest customer:
3. Add items and customer information
4. On Payments: select a Stripe SCA payment
5. Add a card number requiring auth., like the x-3184.
6. After adding the payment check the guest customer email address: no email (nr. 19) arrives.

## Workaround

- No workaround, as these orders are not appearing after customer registration: https://github.com/openfoodfoundation/openfoodnetwork/issues/7905 

## Severity

It concerns payments and there is no workaround for Stripe payments. However, it only happens for BO orders. Subscriptions require customers to be registered.

## Your Environment

* Version used: v.4.1.7
* Browser name and version: All
* Operating System and version (desktop or mobile): All

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?
If the user is set on the order (connected user) send the email to user.email
If this is a guest user, send the email to the order email directly

Closes #8475

As I opened #8882, I cannot test the entire process (I made a hack to test), but I'm confident about the solution I've found.

#### What should we test?

- As a hub manager, create a order for a guest user in the backoffice `/admin/orders/new`
- Create a payment with a card that requires an authorization: `4000 0027 6000 3184`
- User should receive an email
- Repeat the operation with a connected user

#### Release notes

Changelog Category: User facing changes

### CODE REVIEW

File: app/mailers/payment_mailer.rb [hunk 1]
```diff
@@ -7,10 +7,9 @@ def authorize_payment(payment)
     @payment = payment
     subject = I18n.t('spree.payment_mailer.authorize_payment.subject',
                      distributor: @payment.order.distributor.name)
+    to = payment.order.user ? payment.order.user.email : payment.order.email
     I18n.with_locale valid_locale(@payment.order.user) do
-      mail(to: payment.order.user.email,
-           from: from_address,
-           subject: subject)
+      mail(to: to, from: from_address, subject: subject)
```
Comment (Matt-Yorkley) [inline app/mailers/payment_mailer.rb:12]: I'm wondering if there's a good reason to even try both of these values? Could we just go straight for `order.email` instead? We could drop line 10, and just do:

```diff
-- mail(to: to, from: from_address, subject: subject)
++ mail(to: payment.order.email, from: from_address, subject: subject)
```

What do you think?

**Review by Matt-Yorkley — APPROVED** [review]

File: app/mailers/payment_mailer.rb (hunk 1 above)
Comment (jibees) [reply app/mailers/payment_mailer.rb:12]: I thought about it, but didn't want to change to much the code, since I wasn't sure about exactly the consequences (are those two fields well synchronized?).
But for sure, I can take time to be sure that this is well relevant.

Comment (mkllnk) [reply app/mailers/payment_mailer.rb:12]: Good point. I checked the AU production database:
```
Spree::Order.complete.where(email: [nil, ""]).count
=> 0

Spree::Order.joins(:user).complete.where("spree_orders.email != spree_users.email").count
=> 2561
Spree::Order.complete.count
=> 123145
# That's 2%, probably users changing their email at some point.
```

I think that we should simplify it. It should also satisfy Rubocop. :wink:

**Review by mkllnk — COMMENTED** [review]
Great PR!

Let's simplify it though to reduce maintenance cost. :smile:

File: app/mailers/payment_mailer.rb (hunk 1 above)
Comment (jibees) [reply app/mailers/payment_mailer.rb:12]: Just to be sure to understand (I'd be happy to change this ;)): we will change the behavior for 2561 orders.

Thoughts:
 - maybe the user email changed between the time the order has been placed and now?
 - 2% not that much, but not null too

Comment (jibees) [reply app/mailers/payment_mailer.rb]: I've changed that, happy to mark resolved and moved to test ready if that's ok for you both.

**Review by mkllnk — APPROVED** [review]
Great. That's much better. 

I guess that there could be a rare case when someone changed their email and then wants an old confirmation resent... but I don't see why we should support that.

**Comment by filipefurtad0** [comment]
Hey @jibees ,

I've tested this by placing a backoffice order with data for a guest customer.

After adding a payment with a Stripe SCA 3D card, the email now arrives, at the customers account, so I guess this PR closes #8475 :+1: 

There are two scenarios to test:

i) the guest customer is unregistered on OFN

In this case, the customer will need to register first. However, even after doing so, the link received per email does not grant payment authorization, but rather an authorization error, and this Bugsnag JS:

https://app.bugsnag.com/yaycode/open-food-network-js-uk/errors/6203fc5760ddd800087fddae?event_id=621779f400913cfb21920000&i=sk&m=nw

`NotSupportedError: CustomElementRegistry.define: 'turbo-frame' has already been defined as a custom element`

ii) the guest customer is registered on OFN, but just shopping for the first time on that shop

In this case, we're good:
- email arrives and the customer can authorize the payment after logging in

So in conclusion, the PR is a great step in the right direction :muscle: 
But the feature to authorize a payment as a guest customer is not yet really working - issue coming up.
